example.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * This is a hypothetical implementation to understand
+ * what implementations may look like as we define
+ * a standard for video quality of experience metrics.
+ */
+
+// States
+var NOT_PLAYING = 0;
+var ATTEMPTING_TO_PLAY = 1;
+var PLAYING = 2;
+
+
+// Watch Time
+var watchTime = 0;
+var lastWatchTimeCheck;
+var watchTimeInterval;
+
+function updateWatchTime() {
+  watchTime = watchTime + (Date.now() - lastWatchTimeCheck);
+  lastWatchTimeCheck = Date.now();
+}
+
+player.addEventListener('playrequest', function(){
```

Comment:
If `seek` doesn't stop the interval, you'll want to wrap `window.setInterval` in a check to see if you have one running. You'll need to do that anyway, actually, for rebuffering, or else you'll have multiple intervals.